src/validation.h

```diff
@@ -296,7 +289,7 @@ bool UndoReadFromDisk(CBlockUndo& blockundo, const CBlockIndex* pindex);
 bool CheckBlock(const CBlock& block, BlockValidationState& state, const Consensus::Params& consensusParams, bool fCheckPOW = true, bool fCheckMerkleRoot = true);
 
 /** Check a block is completely valid from start to finish (only works on top of our current best block) */
-bool TestBlockValidity(BlockValidationState& state, const CChainParams& chainparams, const CBlock& block, CBlockIndex* pindexPrev, bool fCheckPOW = true, bool fCheckMerkleRoot = true) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+bool TestBlockValidity(BlockValidationState& state, const CChainParams& chainparams, CChainState& chainstate, const CBlock& block, CBlockIndex* pindexPrev, bool fCheckPOW = true, bool fCheckMerkleRoot = true) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
I find these very long lines almost impossible to read. If you touch this branch again, could you be convinced to split it up. Something like:

```suggestion
bool TestBlockValidity(BlockValidationState& state, const CChainParams& chainparams,
                       CChainState& chainstate, const CBlock& block, CBlockIndex* pindexPrev,
                       bool fCheckPOW = true, bool fCheckMerkleRoot = true)
    EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Reply:
Fixed in: d8a5fe24af84c34492e2b23ed341bae70a3c8983